modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php

```diff
@@ -0,0 +1,227 @@
+    // their styles in code.
+    $image_style_setting = $this->getSetting('image_style');
+    if (isset($image_styles[$image_style_setting])) {
+      $summary[] = t('@image_style image style', ['@image_style' => $image_styles[$image_style_setting]]);
+    }
+    else {
+      $summary[] = t('Original image style.');
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item, $langcode);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   * @param string $langcode
+   *   The language code.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item, string $langcode): array {
+    $build = ['#theme' => 'oe_theme_helper_featured_media'];
+    $params = ['description' => $item->caption];
+    $media = $item->entity;
+    $cacheability = CacheableMetadata::createFromRenderArray($build);
+
+    if (!$media instanceof MediaInterface) {
+      return [];
+    }
+
+    // Retrieve the correct media translation.
+    $media = $this->entityRepository->getTranslationFromContext($media, $langcode);
+
+    // Caches are handled by the formatter usually. Since we are not rendering
+    // the original render arrays, we need to propagate our caches to the
+    // oe_theme_helper_featured_media template.
+    $cacheability->addCacheableDependency($media);
+
+    // Get the media source.
+    $source = $media->getSource();
+
+    if ($source instanceof OEmbed) {
+      // Pass it as embeddable data only if it is of type video or html.
+      $oembed_type = $source->getMetadata($media, 'type');
+
+      if (!in_array($oembed_type, ['video', 'html'])) {
+        return [];
+      }
+
+      // Default video aspect ratio is set to 16:9.
+      $params['ratio'] = '16:9';
+      // Load information about the media and the display.
+      $media_type = $this->entityTypeManager->getStorage('media_type')->load($media->bundle());
+      $cacheability->addCacheableDependency($media_type);
+      $source_field = $source->getSourceFieldDefinition($media_type);
+      $display = EntityViewDisplay::collectRenderDisplay($media, 'oe_theme_main_content');
+      $cacheability->addCacheableDependency($display);
+      $display_options = $display->getComponent($source_field->getName());
+      $params['embedded_media'] = $media->{$source_field->getName()}->view($display_options);
+      $build['#params'] = $params;
+      $cacheability->applyTo($build);
+
+      return $build;
+    }
+
+    if ($source instanceof Image) {
+      /** @var \Drupal\image\Plugin\Field\FieldType\ImageItem $thumbnail */
+      $thumbnail = $media->get('thumbnail')->first();
```

Comment:
Why do we use the thumbnail and not the image itself?

Reply:
because of https://github.com/openeuropa/oe_theme/pull/518#discussion_r460762902, https://github.com/openeuropa/oe_theme/blob/a179b359aeb988e534b98d8d832dcacd4c4f8e06/oe_theme.theme#L1033